cmd/swarmctl/service/flagparser/restart.go

```diff
@@ -10,25 +10,26 @@ import (
 )
 
 func parseRestart(flags *pflag.FlagSet, spec *api.ServiceSpec) error {
-	// parse restart-condition
-	condition, err := flags.GetString("restart-condition")
-	if err != nil {
-		return err
-	}
+	if spec.Task.Restart == nil || flags.Changed("restart-condition") {
```

Comment:
@dongluochen Handle it with `--restart-condition` together would set `condition`'s value as `ANY`
Or we could use the code you provided, and change it as follows
```go
if spec.Task.Restart == nil {
            spec.Task.Restart = &api.RestartPolicy{} // Condition is NONE, set by constructor
            spec.Task.Restart.Condition = api.RestartOnAny  // reset it to our default value, ANY
}
```